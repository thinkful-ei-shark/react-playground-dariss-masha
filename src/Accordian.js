import React from 'react';

class Accordian extends React.Component {
    static defaultProps = {
        sections: []
    }

    state = {
        currentIndex: -1
    }

    handleButtonClick(index) {
      console.log(`button clicked, ${index}`)
      this.setState({currentIndex: index})
    }

    render(){
        const buttons = this.props.sections.map((section, index) => 
            ( <li key={index}> 
              <button onClick={() => this.handleButtonClick(index)}>{section.title}</button> 
              {this.state.currentIndex === index && <p>{section.content}</p>}
             </li>)  
        )
        return (
        <ul>
            {buttons}
        </ul>
        )
    }
}

export default Accordian