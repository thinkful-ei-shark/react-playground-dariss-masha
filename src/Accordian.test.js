import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import Accordian from './Accordian';

const testObj = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('Accordian component', () => {
  it('renders without erros', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty ul when no sections prop is supplied', () => {
    const wrapper = shallow(<Accordian />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders no section when no button is clicked', () => {
    const wrapper = shallow(<Accordian sections={testObj} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders section two when button is clicked', () => {
    const wrapper = shallow(<Accordian sections={testObj} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens only the last section when multiple buttons have been clicked', () => {
    const wrapper = shallow(<Accordian sections={testObj} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});