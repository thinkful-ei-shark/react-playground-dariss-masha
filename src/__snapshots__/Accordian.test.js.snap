// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component opens only the last section when multiple buttons have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian component renders empty ul when no sections prop is supplied 1`] = `<ul />`;

exports[`Accordian component renders no section when no button is clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian component renders section two when button is clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
